lib/cloud/imds/gcp/imds.go

```diff
@@ -96,11 +110,7 @@ func (client *InstanceMetadataClient) GetTags(ctx context.Context) (map[string]s
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	idStr, err := client.GetID(ctx)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	id, err := strconv.ParseUint(idStr, 10, 64)
+	id, err := client.getNumericID(ctx)
```

Comment:
So if this fails, are we going to continue to call `GetTags` on a periodic basis and continue to return the same error over and over again?

Reply:
For `getNumericID()` in particular, it's called in `IsAvailable()`, so if it fails (e.g. because we're on Yandex) Teleport won't create a GCP metadata client at all. In general, yes errors here are returned periodically, but we only log them once.

https://github.com/gravitational/teleport/blob/a420d8bb34532fff351ec9b3ce29512901429e49/lib/labels/cloud.go#L147-L158

Reply:
How does this fix #42312?

Reply:
It doesn't, that issue is fixed by the logging changes in `lib/service` (I originally also tried to parse the `*url.Error`s, but per feedback on this PR I removed that as it was redundant).

Reply:
how does the logging change fix that?